components/eamxx/src/share/util/scream_combine_ops.hpp

```diff
@@ -98,6 +100,74 @@ void combine (const ScalarIn& newVal, ScalarOut& result,
       break;
   }
 }
+/* Special version of combine that takes a mask into account */
+template<CombineMode CM, typename ScalarIn, typename ScalarOut,
+         typename CoeffType = typename ekat::ScalarTraits<ScalarIn>::scalar_type>
+KOKKOS_FORCEINLINE_FUNCTION
+void combine_and_fill (const ScalarIn& newVal, ScalarOut& result, const float fill_val,
```

Comment:
We put some effort into making this function generic (meaning with templated types). It seems a bit unsatisfactory to destroy all of that by adding a hard-coded type for the fill value... You could use `ScalarOut`.